### PR TITLE
ci(test): Travis → Circle

### PR DESCRIPTION
**Pull request by Haroenv**
- config file with same setup
- purging unused devDeps
- purging unused scripts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -7,19 +7,14 @@
   "scripts": {
     "watch": "watchify index.js -v -s algoliasearchHelper -o dist/algoliasearch.helper.js",
     "build": "./scripts/build.sh",
-    "dev": "DEBUG=zuul* zuul --no-coverage --local 8090 test/run.js",
```
Comment (Haroenv) [inline package.json]: these didn't work anymore since #624

File: package.json [hunk 2]
```diff
@@ -7,19 +7,14 @@
...
-    "dev-coverage": "DEBUG=zuul* zuul --local 8090 test/run.js",
-    "lint": "scripts/lint.sh",
+    "lint": "eslint .",
     "lint:watch": "onchange 'src/**/*js' 'index.js' 'test/**/*js' '.eslintrc' -- npm run -s lint",
-    "open": "sleep 1 && opn http://localhost:8080/examples",
-    "examples": "npm run build & npm run open & http-server .",
```
Comment (Haroenv) [inline package.json]: these didn't work anymore since the examples moved to codepen, years ago

File: package.json [hunk 3]
```diff
@@ -7,19 +7,14 @@
...
     "doc": "gulp --gulpfile documentation-src/build.gulp.js doc",
     "doc:watch": "gulp --gulpfile documentation-src/build.gulp.js doc:watch",
     "doc:publish": "yarn run doc && gh-pages -d documentation",
     "doc:staging": "scripts/staging-doc.sh",
     "readme:toc": "doctoc --notitle README.md --maxlevel 3",
     "test": "scripts/test.sh",
-    "test:ci": "scripts/test-ci.sh",
```
Comment (Haroenv) [inline package.json]: replaced by circle script

File: package.json [hunk 4]
```diff
@@ -83,25 +75,19 @@
     "metalsmith-serve": "0.0.7",
     "metalsmith-watch": "1.0.3",
     "metalsmith-webpack": "1.0.3",
-    "mkdirp": "0.5.1",
```
Comment (Haroenv) [inline package.json]: couldn't find a usage for this

File: package.json [hunk 5]
```diff
@@ -83,25 +75,19 @@
...
     "mversion": "1.13.0",
     "onchange": "3.2.1",
-    "opn-cli": "3.1.0",
     "pretty-bytes-cli": "2.0.0",
     "prompt": "1.0.0",
     "pug": "^2.0.3",
-    "replace-ext": "1.0.0",
-    "rimraf": "2.6.3",
```
Comment (Haroenv) [inline package.json]: couldn't find usage of these

File: scripts/build.sh [hunk 6]
```diff
@@ -2,7 +2,7 @@
 
 set -e # exit when error
 
-[ -z $TRAVIS_BUILD_NUMBER ] && CI='false' || CI='true' || CI='true'
+[ -z $CIRCLE_BUILD_NUM ] && CI='false' || CI='true' || CI='true'
```
Comment (Haroenv) [inline scripts/build.sh]: not too sure what this line means TBH

File: scripts/test.sh [hunk 7]
```diff
@@ -2,16 +2,10 @@
...
 fi
 
-echo "Test"
-
-echo "Test: node"
-./scripts/test-node.sh
-
-echo "Test: lint"
-./scripts/lint.sh
+node test/run.js
```
Comment (Haroenv) [inline scripts/test.sh]: I wonder if this should just be in package.json too

File: test/integration-spec/helper.distinct.facet.js [hunk 8]
```diff
@@ -7,13 +7,9 @@ var algoliasearchHelper = utils.isCIBrowser ? window.algoliasearchHelper : requi
 
 var random = require('lodash/random');
 
-if (!utils.shouldRun) {
```
Comment (Haroenv) [inline test/integration-spec/helper.distinct.facet.js]: this util was for browser compat, which isn't possible anymore with Jest

File: test/integration-utils.js [hunk 9]
```diff
@@ -34,21 +34,7 @@ function withDatasetAndConfig(indexName, dataset, config) {
...
-if (!process.browser) {
-  shouldRun = true;
-} else if ('XDomainRequest' in window) {
-  shouldRun = false;
-} else {
-  shouldRun = true;
-}
-
 module.exports = {
-  isCIBrowser: process.browser && process.env.TRAVIS_BUILD_NUMBER,
```
Comment (Haroenv) [inline test/integration-utils.js]: this was for browser compat

File: scripts/build.sh (hunk 6 above)
Comment (samouss) [inline scripts/build.sh]:
```suggestion
[ -z $CIRCLE_BUILD_NUM ] && CI='false' || CI='true'
```

File: .circleci/config.yml [hunk 10]
```diff
@@ -0,0 +1,75 @@
...
+      - checkout
+      - run: *install_yarn_version
+      - restore_cache: *restore_yarn_cache
+      - run: *run_yarn_install
+      - save_cache: *save_yarn_cache
+      - run:
+          name: Lint & Code styles
+          command: yarn lint
+      - run:
+          name: Unit Tests
```
Comment (samouss) [inline .circleci/config.yml]: It's not only the unit tests we run both units & integrations at the same time. 

We also have to update the name of the job.

File: .circleci/config.yml [hunk 11]
```diff
@@ -0,0 +1,75 @@
...
+    command: yarn install
+
+defaults: &defaults
+  working_directory: ~/algoliasearch-helper-js
+  docker:
+    - image: circleci/node:8.15.1@sha256:3d32b8457083c33c04eb60fac02c0e39de5bd28de4b95393d7dcb5aa37b20af7
+
+version: 2
+jobs:
+  test_build:
```
Comment (samouss) [inline .circleci/config.yml]:
```suggestion
  build:
```

File: .circleci/config.yml [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+      - run: *run_yarn_install
+      - save_cache: *save_yarn_cache
+      - run:
+          name: Lint & Code styles
+          command: yarn lint
+      - run:
+          name: Unit & Integration Tests
+          command: yarn test --maxWorkers=4
+
+  test_docs:
```
Comment (samouss) [inline .circleci/config.yml]:
```suggestion
  docs:
```

**Pull request merged**